### ISSUE TITLE
vcpkg install under macOS

### ISSUE THREAD

**Issue by niyue** [etcd-cpp-apiv3/etcd-cpp-apiv3#68]
I tried to install etcd-cpp-apiv3 via vcpkg (but under macOS instead of Windows), and ran into the same issue as #64
```
cd /Users/ss/dev/tools/vcpkg/buildtrees/etcd-cpp-apiv3/x64-osx-dbg/proto && /Users/ss/dev/tools/vcpkg/installed/x64-osx/tools/protobuf/protoc --cpp_out /Users/ss/dev/tools/vcpkg/buildtrees/etcd-cpp-apiv3/x64-osx-dbg/proto/gen/proto -I /Users/ss/dev/tools/vcpkg/buildtrees/etcd-cpp-apiv3/src/v0.2.1-85b6180791.clean/proto /Users/ss/dev/tools/vcpkg/buildtrees/etcd-cpp-apiv3/src/v0.2.1-85b6180791.clean/proto/auth.proto
google/protobuf/descriptor.proto: File not found.
gogoproto/gogo.proto:32:1: Import "google/protobuf/descriptor.proto" was not found or had errors.
gogoproto/gogo.proto:38:8: "google.protobuf.EnumOptions" is not defined.
gogoproto/gogo.proto: "google.protobuf.EnumOptions" is not defined.
gogoproto/gogo.proto: "google.protobuf.EnumOptions" is not defined.
```

I used the vcpkg portfile in this repo, and grpc/protobuf is installed into `$VCPKG_ROOT/installed/x64-osx` directory (`descriptor.proto` can be found under `$VCPKG_ROOT/installed/x64-osx/include/google/protobuf`), what is the recommended approach to add this path for the build? Thanks.

**Comment by niyue** [comment]
I saw @sighingnow's comment in #64 saying 
> If you want to install grpc and protobuf to a customized directory, you could specify the CMAKE_INSTALL_PREFIX.

But I am not sure if vcpkg's install path is considered as a `customized directory` since I simply want to use the dependencies (grpc/protobuf) installed in vcpkg.

**Comment by sighingnow** [comment]
Could you please check your vcpkg version and installed protobuf version? I have previously reported a bug in vcpkg and it has already been fixed, see https://github.com/microsoft/vcpkg/issues/15217 and https://github.com/microsoft/vcpkg/pull/15246.

I also made some vcpkg integration there: https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3/tree/master/vcpkg/etcd-cpp-apiv3, and it should still work (haven't tested it on master yet).

**Comment by niyue** [comment]
To ensure this is not a vcpkg issue, I upgrade it to the latest revision today, the current version is:
```
❯ vcpkg version
Vcpkg package management program version 2021-05-05-unknownhash
```

For protobuf, its version is 3.13.0#2:
```
❯ vcpkg list | grep protobuf
protobuf:x64-osx                                   3.13.0#2         Protocol Buffers - Google's data interchange format
```

**Comment by sighingnow** [comment]
As in this patch https://github.com/microsoft/vcpkg/pull/15246/files the protobuf version is 3.14, I think you may need to upgrade the protobuf installation first.

**Comment by niyue** [comment]
Let me give it a try and will get back to you later.

**Comment by niyue** [comment]
@sighingnow upgrading protobuf works for me. I upgraded both protobuf and grpc to the latest version in vcpkg:
```
❯ vcpkg list | grep -e protobuf -e grpc
grpc:x64-osx                                       1.37.0#2         An RPC library and framework
grpc[codegen]:x64-osx                                               Build code generator machinery
protobuf:x64-osx                                   3.15.8#2         Protocol Buffers - Google's data interchange format
```

And after upgrading, the installation of etcd-cpp-apiv3 worked partially. vcpkg reported error like below:
```
-- Building x64-osx-dbg
-- Building x64-osx-rel
-- Installing: /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/share/etcd-cpp-apiv3/copyright
-- Performing post-build validation
The /lib/cmake folder should be merged with /debug/lib/cmake and moved to /share/etcd-cpp-apiv3/cmake.
Please use the helper function `vcpkg_fixup_cmake_targets()`
The following cmake files were found outside /share/etcd-cpp-apiv3. Please place cmake files in /share/etcd-cpp-apiv3.

    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/lib/cmake/etcd-cpp-api/etcd-targets-release.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/lib/cmake/etcd-cpp-api/etcd-cpp-api-config.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/lib/cmake/etcd-cpp-api/etcd-targets.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/lib/cmake/etcd-cpp-api/etcd-cpp-api-config-version.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/lib/cmake/etcd-cpp-api/FindGRPC.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/debug/lib/cmake/etcd-cpp-api/etcd-cpp-api-config.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/debug/lib/cmake/etcd-cpp-api/etcd-targets.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/debug/lib/cmake/etcd-cpp-api/etcd-targets-debug.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/debug/lib/cmake/etcd-cpp-api/etcd-cpp-api-config-version.cmake
    /Users/ss/dev/tools/vcpkg/packages/etcd-cpp-apiv3_x64-osx/debug/lib/cmake/etcd-cpp-api/FindGRPC.cmake

The /debug/lib/cmake folder should be merged with /lib/cmake into /share/etcd-cpp-apiv3
Found 3 error(s). Please correct the portfile:
    /Users/ss/dev/projects/dl/stonewave/ci/dev_image/ports/etcd-cpp-apiv3/portfile.cmake
```

I had to add some additional cmake instructions to make it work in etcd-cpp-apiv3's `vcpkg/portfile.cmake`:
```
vcpkg_install_cmake()
vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/etcd-cpp-api) # newly added

file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share) # newly added

vcpkg_copy_pdbs()
```

After this fix, the vcpkg installation worked for me on macOS.

Let me know if this is the expected fix, and I am glad to submit a PR if this is needed.

**Comment by sighingnow** [comment]
> After this fix, the vcpkg installation worked for me on macOS.

Thanks a lot! PR is very welcome!

**Comment by niyue** [comment]
Please check it out here, thanks.
https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3/pull/69

**Comment by sighingnow** [comment]
Closed by pull request #69.

**Comment by niyue** [comment]
@sighingnow thanks so much for your timely response!
